Islandora/islandora_defaults
# Use View Mode Switcher.

**GitHub Issue**: [Use-a-thon](https://docs.google.com/document/d/1Sw0CiOF_yavYgAs2XysK25D9_Ccc5798QskM_figf6o/edit#): Simplify how viewers get selected and displayed


# What does this Pull Request do?

Makes "View Mode Switch" a requirement.
Creates a "Default Viewer" view mode for media
Configures this Default Viewer view mode to show all media in a sensible default viewer
Sets up a View Mode Switch field on Image, Document, and File
Lets you set the field on the media to switch the view mode to override Default:
- to OpenSeadragon on Image and File
- to PDFjs on File and Document
Removes the PDFjs and OpenSeadragon EVA views
Removes the PDFjs, OpenSeadragon, and Binary view modes from nodes
Removes the PDFjs, OpenSeadragon, and Binary contexts (that used the node view modes) 


* Does this change require documentation to be updated?  - yep! But not too much.
* Does this change add any new dependencies?  - yes. Included in composer.json. it's in security coverage.
* Does this change require any other modifications to be made to the repository
 (i.e. Regeneration activity, etc.)? n/a
* Could this change impact execution of existing code? no

# How should this be tested?

Install this PR on a fresh (never seen islandora defaults) site.
Note that this means display is **independent** of the node's islandora model.
Note that this means the file that will display is the Service file! Double-tag if you're not making a separate Service File.

Ensure that:
- an object that has an Image (standard Image) Service File will show that service file on the node
- if you use the View Mode Switcher you can make that image show up in openseadragon
- Similarly with PDFs in Documents. 
- Similarly with PDFs or TIFFs in Files.


# Additional Notes:
Any additional information that you think would be helpful when reviewing this
 PR.

# Interested parties
@Islandora/committers
